circuits/circom/core/non-qv/processMessages.circom

```diff
@@ -187,6 +189,9 @@ include "../../trees/incrementalQuinaryTree.circom";
     var derivedPubKey[2] = PrivToPubKey()(coordPrivKey);
     derivedPubKey === coordPubKey;
```

Comment:
you can get rid of this check and the `coordPubKey` signal - you are already checking that the the hash of the derived key is = to the hash that is on chain